src/repo.rs

```diff
@@ -341,6 +341,13 @@ impl Repository {
         Ok(repo)
     }
 
+    /// Clone a remote repository with a depth of one (shallow cloning)
+    ///
+    /// Similar to `git clone --depth 1`
+    pub fn shallow_clone<P: AsRef<Path>>(url: &str, into: P) -> Result<Repository, Error> {
```

Comment:
Similar to the comment on `RepoBuilder`, I think I would like to avoid having too many convenience functions, since there is a somewhat combinatorial explosion of all the different options someone might want. Using a builder pattern can avoid adding all sorts of variants of different methods.